crates/driver/src/domain/competition/mod.rs

```diff
@@ -413,6 +512,13 @@ fn merge(solutions: impl Iterator<Item = Solution>, auction: &Auction) -> Vec<So
     merged
 }
 
+struct SettleRequest {
+    auction_id: Option<i64>,
+    solution_id: u64,
+    submission_deadline: u64,
+    response_sender: oneshot::Sender<Result<Settled, Error>>,
+}
```

Comment:
Can we use domain types here? `eth::BlockNo` for block numbers, `solution::Id` for solutions id, `auction::Id` for auction id?

Reply:
Migrated to domain types, except `solution:Id` since it's `::new()` function generates a new id under the hood and constructing directly from `u64` seems like incorrect.